### DISCUSSION TITLE
popover's data-content cannot parse such tags as <button><input>

### DISCUSSION THREAD

**Discussion by cyxhub** [twbs/bootstrap#32269] [Q&A] [answered]
Why can popover's data-content parse <b>, <i>, etc. But it cannot parse such tags as &lt;button&gt;,&lt;input&gt;, etc.
This is my confusion.

`<button type="button" class="btn btn-lg btn-danger" data-html="true" data-toggle="popover" title="Popover title" data-content="<b>bold</b> <input type='text'/> <button>btn</button> ">Click to toggle popover</button>
`

**Comment by patrickhlauke** [comment, marked as answer]
https://getbootstrap.com/docs/4.5/getting-started/javascript/#sanitizer

further, i'd also recommend looking at the "Making popovers work for keyboard and assistive technology users" note in https://getbootstrap.com/docs/4.5/components/popovers/#usage though

> Additionally, while it is possible to also include interactive controls (such as form elements or links) in your popover (by adding these elements to the `whiteList` or allowed attributes and tags), be aware that currently the popover does not manage keyboard focus order. When a keyboard user opens a popover, focus remains on the triggering element, and as the popover usually does not immediately follow the trigger in the document’s structure, there is no guarantee that moving forward/pressing `TAB` will move a keyboard user into the popover itself. In short, simply adding interactive controls to a popover is likely to make these controls unreachable/unusable for keyboard users and users of assistive technologies, or at the very least make for an illogical overall focus order. In these cases, consider using a modal dialog instead.

**Reply by cyxhub** [reply]
Thank you very much. I have understood it.
